CRM/Volunteer/BAO/Project.php

```diff
@@ -250,10 +250,22 @@ public static function create(array $params) {
 
     $project->save();
 
+    // Prevent updates for unpermissioned users.
+    $hasEditVPC = CRM_Volunteer_Permission::check('edit volunteer project relationships');
+    $hasEditProfiles = CRM_Volunteer_Permission::check('edit volunteer registration profiles');
```

Comment:
I don't think so. I didn't like checking the permissions more than once, which is in part what you complained about above, but the checks serve different purposes.

In `supplyDefaults` they decide whether to honor the user submission or force the defaults. In `create` they determine whether to write the ancillary entities at all.